perma_web/perma/forms.py

```diff
@@ -516,16 +512,16 @@ def clean_csv_file(self):
             except ValidationError:
                 raise forms.ValidationError(f"CSV file contains invalid email address: {email}")
 
-            if email in seen:
+            if email.lower() in self.user_data:
                 raise forms.ValidationError(f"CSV file cannot contain duplicate users: {email}")
             else:
-                seen.add(email)
-                self.user_data[email] = {
+                self.user_data[email.lower()] = {
+                    'raw_email': email,
                     'first_name': row.get('first_name', '').strip(),
                     'last_name': row.get('last_name', '').strip()
                 }
 
-        if row_count == 0:
+        if not len(self.user_data):
```

Comment:
This is trivial, but you could simplify the condition even further to:
```suggestion
        if not self.user_data:
```

Reply:
Ha! Good point, not sure what happened mentally there :-)